pandas/tests/extension/integer/test_integer.py

```diff
@@ -143,283 +129,77 @@ def test_array_type_with_arg(self, data, dtype):
-        mask = s.isna()
-
-        # other array is an Integer
-        if isinstance(other, IntegerArray):
-            omask = getattr(other, 'mask', None)
-            mask = getattr(other, 'data', other)
-            if omask is not None:
-                mask |= omask
-
-        # float result type or float op
-        if ((is_float_dtype(other) or is_float(other) or
-             op_name in ['__rtruediv__', '__truediv__',
-                         '__rdiv__', '__div__'])):
-            rs = s.astype('float')
-            expected = op(rs, other)
-            self._check_op_float(result, expected, mask, s, op_name, other)
-
-        # integer result type
-        else:
-            rs = pd.Series(s.values._data)
-            expected = op(rs, other)
-            self._check_op_integer(result, expected, mask, s, op_name, other)
-
-    def _check_op_float(self, result, expected, mask, s, op_name, other):
-        # check comparisions that are resulting in float dtypes
-
-        expected[mask] = np.nan
-        self.assert_series_equal(result, expected)
-
-    def _check_op_integer(self, result, expected, mask, s, op_name, other):
-        # check comparisions that are resulting in integer dtypes
-
-        # to compare properly, we convert the expected
-        # to float, mask to nans and convert infs
-        # if we have uints then we process as uints
-        # then conert to float
-        # and we ultimately want to create a IntArray
-        # for comparisons
-
-        fill_value = 0
-
-        # mod/rmod turn floating 0 into NaN while
-        # integer works as expected (no nan)
-        if op_name in ['__mod__', '__rmod__']:
-            if is_scalar(other):
-                if other == 0:
-                    expected[s.values == 0] = 0
-                else:
-                    expected = expected.fillna(0)
+    def check_opname(self, s, op_name, other, exc=None):
+        # overwriting to indicate ops don't raise an error
+        super(TestArithmeticOps, self).check_opname(s, op_name,
+                                                    other, exc=None)
+
+    def _check_op(self, s, op, other, op_name, exc=NotImplementedError):
+        if exc is None:
+            if s.dtype.is_unsigned_integer and (op_name == '__rsub__'):
+                # TODO see https://github.com/pandas-dev/pandas/issues/22023
+                pytest.skip("unsigned subtraction gives negative values")
+
+            result = op(s, other)
+            expected = s.combine(other, op)
+
+            if op_name in ('__rtruediv__', '__truediv__'):
+                expected = expected.astype(float)
+                if op_name == '__rtruediv__':
+                    # TODO reverse operators result in object dtype
+                    result = result.astype(float)
+            elif op_name.startswith('__r'):
+                # TODO reverse operators result in object dtype
+                # see https://github.com/pandas-dev/pandas/issues/22024
```

Comment:
some of the added checks here have to do with certain 'buggy' (or unexpected) behaviours (see the issues I opened)

Reply:
if you are doing a PR to move things, don't change code. I have no idea what you are changing here.

Reply:
Well, the original code is still in `tests/arrays/test_integer.py`, so I only moved it there. 
The changes here are simply to get the base class tests working instead of overwriting them as you did (and again, those custom tests are still in the other file).